cmd/verifier/e2e_test.go

```diff
@@ -0,0 +1,216 @@
+	if err != nil {
+		t.Errorf("error getting log info: %v", err)
+	}
+	checkpoint := &util.SignedCheckpoint{}
+	if err := checkpoint.UnmarshalText([]byte(*logInfo.SignedTreeHead)); err != nil {
+		t.Errorf("%v", err)
+	}
+	iterator := 0
+	for checkpoint.Size <= 0 {
+		logInfo, err = rekor.GetLogInfo(context.Background(), rekorClient)
+		if err != nil {
+			t.Errorf("error getting log info: %v", err)
+		}
+		checkpoint := &util.SignedCheckpoint{}
+		if err := checkpoint.UnmarshalText([]byte(*logInfo.SignedTreeHead)); err != nil {
+			t.Errorf("error unmarshalling checkpoint: %v", err)
+		}
+		iterator++
+		if iterator >= 5 {
+			t.Errorf("log info checkpoint failed to update in time")
+		}
+		time.Sleep(2 * time.Second)
+	}
+
+	tempDir := t.TempDir()
+	tempLogInfoFile, err := os.CreateTemp(tempDir, "")
+	if err != nil {
+		t.Errorf("failed to create temp log file: %v", err)
+	}
+	tempLogInfoFileName := tempLogInfoFile.Name()
+	defer os.Remove(tempLogInfoFileName)
+
+	tempOutputIdentitiesFile, err := os.CreateTemp(tempDir, "")
+	if err != nil {
+		t.Errorf("failed to create temp output identities file: %v", err)
+	}
+	tempOutputIdentitiesFileName := tempOutputIdentitiesFile.Name()
+	defer os.Remove(tempOutputIdentitiesFileName)
+
+	interval := time.Minute
+	monitoredVals := identity.MonitoredValues{
+		Subjects: []string{subject},
+	}
+	once := true
+
+	err = RunConsistencyCheck(&interval, rekorClient, verifier, &tempLogInfoFileName, monitoredVals, &tempOutputIdentitiesFileName, &once)
+	if err != nil {
+		t.Errorf("first consistency check failed: %v", err)
+	}
+
+	payload = []byte{1, 2, 3, 4, 5, 6}
+	sig, err = signer.SignMessage(bytes.NewReader(payload))
+	if err != nil {
+		t.Fatalf("error signing message: %v", err)
+	}
+	hashedrekord = &hashedrekord_v001.V001Entry{}
+	hash = sha256.Sum256(payload)
+	pe, err = hashedrekord.CreateFromArtifactProperties(context.Background(), types.ArtifactProperties{
+		ArtifactHash:   hex.EncodeToString(hash[:]),
+		SignatureBytes: sig,
+		PublicKeyBytes: [][]byte{pemCert},
+		PKIFormat:      "x509",
+	})
+	if err != nil {
+		t.Fatalf("error creating hashed rekord log entry: %v", err)
+	}
+	params = entries.NewCreateLogEntryParams()
+	params.SetProposedEntry(pe)
+	resp, err = rekorClient.Entries.CreateLogEntry(params)
+	if !resp.IsSuccess() || err != nil {
+		t.Errorf("error creating log entry: %v", err)
+	}
+	err = RunConsistencyCheck(&interval, rekorClient, verifier, &tempLogInfoFileName, monitoredVals, &tempOutputIdentitiesFileName, &once)
+	if err != nil {
+		t.Errorf("second consistency check failed: %v", err)
+	}
+
+	tempOutputIdentities, err := os.ReadFile(tempOutputIdentitiesFileName)
+	if err != nil {
+		t.Errorf("error reading from output identities file: %v", err)
+	}
+	tempOutputIdentitiesString := string(tempOutputIdentities)
+	if !strings.Contains(tempOutputIdentitiesString, subject) {
+		t.Errorf("expected to find %s, did not", subject)
+	}
+}
```

Comment:
For identities in certificates, the issuer should be included - https://github.com/sigstore/rekor-monitor/blob/c3b66b93b2bdb556b986e6a2c22470270c7fc697/pkg/rekor/identity.go#L101

The `Subjects` field is somewhat confusingly named - this includes references to identifiers found in everything but certificates, like how PGP keys have emails.

Reply:
Actually, I think this would be a good thing to test for as well - record a key and check for its fingerprint, check only a subject, check a subject+issuer, and check for an OID match.